### PR TITLE
Raise minimum JDK version to 17

### PR DESCRIPTION
**Pull request by J-N-K**
Depends on https://github.com/openhab/openhab-core/pull/2994

Signed-off-by: Jan N. Klug <github@klug.nrw>

### CODE REVIEW

**Review by hmerk — APPROVED** [review]
LGTM

File: itests/itest-common.bndrun [hunk 1]
```diff
@@ -19,7 +19,7 @@ Test-Cases: ${classes;CONCRETE;PUBLIC;NAMED;*Test}
 -runsystempackages: sun.reflect
 
 -runfw: org.eclipse.osgi
--runee: JavaSE-11
```
Comment (wborn) [inline itests/itest-common.bndrun:22]: Can you also remove `nashorn.args=--no-deprecation-warning,\` on line 42?

File: pom.xml [hunk 2]
```diff
@@ -384,7 +385,7 @@ Import-Package: \\
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-surefire-plugin</artifactId>
-          <version>3.0.0-M5</version>
+          <version>3.0.0-M6</version>
```
Comment (wborn) [inline pom.xml:388]: If it works, I would use:

```suggestion
          <version>3.0.0-M7</version>
```

**Review by wborn — APPROVED** [review]
Thanks!

**Comment by wborn** [comment]
Can you fix the build  @J-N-K? I think CI for this PR succeeded but now it no longer does because of dependency changes. Also the GHA Java 11 build still needs to be removed.

**Comment by J-N-K** [comment]
I‘ll have a look later this evening.

**Pull request merged**